### PR TITLE
fix(authproxy): cookie secret creation and reference using env valueFrom

### PR DESCRIPTION
**Pull request by elias-gb**
### Fix

- Added relevant cookie secret.
- Added test for cookie secret.

## Motivations 
There was no cookie secret to be looked up in subsequent helm upgrade.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (andrewazores) [inline .gitignore:1]: Please configure yourself a global gitignore on your workstation, and put these IDE-specific files and directories there instead.

**Review by andrewazores — COMMENTED** [review]
Hi,

Thanks for the contribution! Could you detail how to reproduce the bug you ran into? It sounds like you did an install and then an upgrade, and that's when you hit a problem with the Secret?

**Comment by elias-gb** [comment]
> Hi,
> 
> Thanks for the contribution! Could you detail how to reproduce the bug you ran into? It sounds like you did an install and then an upgrade, and that's when you hit a problem with the Secret?

There is no secrets created for cookies during both the initial install and upgrades. We noticed the issue as you described when we tried to install and upgrade. Each time it tried to recreate the cookie secrets on authproxy.

**Comment by andrewazores** [comment]
@elias-gb could you test the latest commit I pushed and ensure it works on your setup?

The Secret you've defined should get properly created on installation still, but it will get recreated with a new secret value on upgrade. This shouldn't be too much of a problem, it only means that after upgrade any users' browsers will have an invalid session and will need to re-login. That was already the case before this PR anyway since the Secret was never actually created. At least now the Secret is properly created and referenced in a better way.

**Comment by elias-gb** [comment]
> @elias-gb could you test the latest commit I pushed and ensure it works on your setup?
> 
> ~The Secret you've defined should get properly created on installation still, but it will get recreated with a new secret value on upgrade. This shouldn't be too much of a problem, it only means that after upgrade any users' browsers will have an invalid session and will need to re-login. That was already the case before this PR anyway since the Secret was never actually created. At least now the Secret is properly created and referenced in a better way.~

Tested it now. Looks fine :)

File: charts/cryostat/templates/cookie_secret.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: {{ .Release.Name }}-cookie-secret
+type: Opaque
+{{ if $secret -}}
+data:
+  COOKIE_SECRET: {{ $secret.data.COOKIE_SECRET }}
+{{ else -}}
+stringData:
+  COOKIE_SECRET: {{ randAlphaNum 32 }}
+{{ end }}
```
Comment (tthvo) [inline charts/cryostat/templates/cookie_secret.yaml:13]: How about keeping the "cryostat.cookieSecret" and reference it here in the `data` field as in other secrets that helm creates?

```go
{{/*
Generate or retrieve a default value for cookieSecret.
*/}}
{{- define "cryostat.cookieSecret" -}}
{{- $secret := (lookup "v1" "Secret" .Release.Namespace (printf "%s-cookie-secret" .Release.Name)) -}}
{{- if $secret -}}
{{/*
   Use the current secret. Do not regenerate.
*/}}
{{- $secret.data.COOKIE_SECRET -}}
{{- else -}}
{{/*
    Generate a new secret.
*/}}
{{- randAlphaNum 32 | b64enc | quote -}}
{{- end }}
{{- end }}
```

**Review by tthvo — COMMENTED** [review]
Tested both on k8s (minikube) and OpenShift Local. Looks good^^

**Review by ebaron — APPROVED** [review]
Looks good to me, thank you for your contribution @elias-gb!

**Pull request merged**